public/app/plugins/datasource/grafana-azure-monitor-datasource/components/VariableEditor/VariableEditor.test.tsx

```diff
@@ -82,6 +82,43 @@ describe('VariableEditor:', () => {
     });
   });
 
+  describe('Azure Resource Graph queries:', () => {
+    const ARGquery = {
+      refId: 'A',
+      queryType: AzureQueryType.AzureResourceGraph,
+      azureResourceGraph: {
+        query: 'Resources | distinct type',
+      },
+      subscriptions: ['Subscription1', 'Subscription2'],
+    };
+
+    it('should render', async () => {
+      render(<VariableEditor {...defaultProps} query={ARGquery} />);
+      await waitFor(() => screen.queryByTestId('mockeditor'));
+      expect(screen.queryByLabelText('Subscriptions')).toBeInTheDocument();
+      expect(screen.queryByLabelText('Select subscription')).not.toBeInTheDocument();
+      expect(screen.queryByLabelText('Select query type')).not.toBeInTheDocument();
+      expect(screen.queryByLabelText('Select resource group')).not.toBeInTheDocument();
+      expect(screen.queryByLabelText('Select namespace')).not.toBeInTheDocument();
+      expect(screen.queryByLabelText('Select resource')).not.toBeInTheDocument();
```

Comment:
Not ideal but I could not find a better way to test that the other fields are not rendered.

Reply:
I would at least check that `Select Subscription` is not rendered as it's an easy mistake ; we have `subscriptions` plural for ARG, so multitag select, but not the dropdwn the other query functions have.